packages/i18n-calypso/src/number-formatters/get-cached-formatter.ts

```diff
@@ -0,0 +1,40 @@
+const formatterCache = new Map();
+
+interface Params {
+	locale: string;
+	options?: Intl.NumberFormatOptions;
+	fallbackLocale?: string;
+	retries?: number;
+}
+
+export function getCachedFormatter( {
+	locale,
+	fallbackLocale = 'en',
+	options,
+	retries = 1,
+}: Params ): Intl.NumberFormat {
+	const cacheKey = JSON.stringify( [ locale, options ] );
+
+	try {
+		return (
+			formatterCache.get( cacheKey ) ??
+			formatterCache.set( cacheKey, new Intl.NumberFormat( locale, options ) ).get( cacheKey )
+		);
+	} catch ( error ) {
+		// If the locale is invalid, creating the NumberFormat will throw.
+		// eslint-disable-next-line no-console
+		console.warn(
+			`Intl.NumberFormat was called with a non-existent locale "${ locale }"; falling back to ${ fallbackLocale }`
+		);
+
+		if ( retries ) {
+			return getCachedFormatter( {
+				locale: fallbackLocale,
+				options,
+				retries: retries - 1,
+			} );
+		}
+
+		throw error;
```

Comment:
Added a limit on reitres so it doesn't end up in an infinite loop - theoretically, as that's unlikely to happen with a valid fallback locale. Also failing loudly if for whatever reason it happens.